process/block/poolsCleaner/crossTxsPoolsCleaner.go

```diff
@@ -115,7 +128,19 @@ func (ctpc *crossTxsPoolsCleaner) receivedRewardTx(key []byte, value interface{}
 	}
 
 	log.Trace("crossTxsPoolsCleaner.receivedRewardTx", "hash", key)
-	ctpc.processReceivedTx(key, value, rewardTx)
+
+	tx, ok := value.(data.TransactionHandler)
+	if !ok {
+		log.Warn("crossTxsPoolsCleaner.receivedRewardTx", "error", process.ErrWrongTypeAssertion)
+		return
+	}
+
+	senderShardID, receiverShardID, err := ctpc.computeSenderAndReceiverShards(tx)
+	if err != nil {
+		log.Debug("crossTxsPoolsCleaner.receivedRewardTx", "error", err.Error())
```

Comment:
shouldn't you return here on error as well?

Reply:
Yep, forgot to do it